### ISSUE TITLE
New line added if import is in function

### ISSUE THREAD

**Issue by shardros** [timothycrosley/isort#1277]
Hello,

Firstly thank you for all your work on `isort` v5 is truely something.

I know this is a bit nasty to be doing anyway but if your imports are in a function with one line between them `isort` will always add a new line between them.

Before `isort`:
```py
def main():
    import time

    import sys
```

After `isort`:
```py
def main():
    import time


    import sys
```

After running `isort` a second time:

```py
def main():
    import time



    import sys
```

Tested on commit: 686d3c715183eb8ba944475512e60342cadabaf6

**Comment by shardros** [comment]
The current work around which I am using is to run autopep8 after isort which cleans this up.

**Comment by timothycrosley** [comment]
Hi @shardros,

Thanks for the positive feedback on the 5.0.0 release, it is really appreciated!

I'm sorry you are experiencing this issue, and I hope I can resolve it as quickly as possible. I haven't been able to reproduce it locally or on CI/CD, but I trust with some additional info I will be able to. 

Could you
- [ ] Try isort 5.0.5 and see if it still occurs for you? (there's a small chance this was fixed in develop before I added the test case)
- [ ] Provide any settings used when running isort. A quick way to get this is to paste the result of `isort --show-config`
- [ ] Provide the OS isort is being ran from?

Thanks!

~Timothy

**Comment by ziima** [comment]
I can't quite reproduce the issue, but I have a similar issue. Isort splits imports in functions in sections, which I don't really like.

Running
```
isort --check-only --diff test.py
```
on a file
```python
def main():
    import sys
    from django import setup
```
produces
```
--- /tmp/test/test.py:before	2020-07-09 09:09:32.168479
+++ /tmp/test/test.py:after	2020-07-09 09:09:44.959117
@@ -1,3 +1,4 @@
 def main():
     import sys
+
     from django import setup
```

In my opinion, splitting the imports into sections inside functions is not that useful. I haven't found a setting which would help with that.

**Comment by shardros** [comment]
Hello @timothycrosley,

Thank you for your suggestions. [Here is a repository](https://github.com/shardros/isort-bug-demo) with what I believe to be minimal code needed to reproduce the bug.

This was tested on windows. I don't have access to a Linux system to test this on right now but will try over the weekend. The output of `isort --show-config` and the exact OS is listed in the `README.md` of the repo. 

Sadly `isort 5.0.5` does not seem to solve the problem. 

Thank you for all of your hard work,

Edwin

**Comment by shardros** [comment]
I am unable to reproduce @ziima in the virtual env in [this repository](https://github.com/shardros/isort-bug-demo) which exhibits the original bug.

**Comment by ziima** [comment]
You have constrained isort version to 5.0.5. There is already 5.0.7.

**Comment by sbrugman** [comment]
Thanks Timothy for your work on v5, much appreciated!

In case it might help locating the issue: we observe the same behaviour when running Windows and Ubuntu through the WSL. Isort works as expected on Github Actions running Ubuntu.

**Comment by timothycrosley** [comment]
Thanks everyone for the detailed information on this issue! Based on the details I borrowed a windows machine and was able to reproduce the issue almost right away. This should be fixed in the just release 5.0.8 release, with an additional regression test added to ensure it doesn't reappear.

**Comment by shardros** [comment]
Thank you @timothycrosley
